traversal/selector/helpers.go

```diff
@@ -0,0 +1,35 @@
+package selector
+
+import (
+	"strings"
+
+	"github.com/ipld/go-ipld-prime/codec/json"
+	"github.com/ipld/go-ipld-prime/node/basic"
+)
+
+// REVIEW: I feel sketchy about CompileJSONSelector because it brings in json as a transitive dependency.
+// This has consequences as far as dragging go-cid in because our json shares code with dagjson...
+// and then go-cid ends up in the transitives of the traversal package, since traversal depends on selector.
+// That's... not great.
+// On the other hand: trying to keep up that paper wall is creating a lot of strife.  Maybe it's time to give up.
+
+func CompileJSONSelector(jsonStr string) (Selector, error) {
```

Comment:
I think there'd be a strong argument for calling this `ParseJSONAndCompileSelector`, but of course that's a mouthful.